mempool/s3_memory_pool.c

```diff
@@ -330,6 +337,12 @@ void *mempool_getbuffer(MemoryPoolHandle handle, size_t expected_buffer_size) {
   if (pool_item) {
     pool->number_of_bufs_shared++;
   }
+  snprintf(log_mem_stats, sizeof(log_mem_stats), log_memool_stats,
+           pool->mempool_item_size, pool->free_bufs_in_pool,
+           pool->number_of_bufs_shared, pool->total_bufs_allocated_by_pool);
```

Comment:
maybe move it into the `if` below?  This buffer is not used if function is not defined.